libvast/src/io/save.cpp

```diff
@@ -21,31 +21,33 @@
 #include <cstddef>
 #include <cstdio>
 #include <filesystem>
+#include <string>
 #include <system_error>
 
 namespace vast::io {
 
-caf::error save(const path& filename, span<const std::byte> xs) {
-  auto tmp = filename + ".tmp";
-  if (auto err = write(tmp, xs)) {
+caf::error
+save(const std::filesystem::path& filename, span<const std::byte> xs) {
+  auto tmp = std::filesystem::path{filename.string() + ".tmp"};
```

Comment:
```suggestion
  auto tmp = filename + ".tmp";
```

See https://en.cppreference.com/w/cpp/filesystem/path/concat.

(Not sure if I missed similar instances of this in the past.)

Reply:
We can't use `concat` or `operator+=` as it mutates the LHS operand. In this case, we need two `std::filesystem::path` variables: the original `filename` and then a new `tmp` file. We can't modify `filename` as we need it for the rename later. Do you see something I'm missing?

Reply:
Interesting. Now that I looked closely, there is no `operator+` while there exists an `operator/`. Apparently the committee deems it to risky. 🤷 

So yeah, there are basically two ways to concatenate raw paths. Either do it the way you suggested or potentially save an allocation by doing this:

```cpp
auto tmp = filename;
filename += ".tmp";
```

:point_up: This has less characters overall, so while it expands more vertically, it's quicker to understand. However, I don't feel strongly about this change. After all, we're doing I/O immediately afterwards.

Reply:
I just went ahead with your proposed change; a bit shorter and easier to reason about I'd say. 👍